bindings/matrix-sdk-ffi/src/authentication_service.rs

```diff
@@ -523,23 +469,24 @@ impl AuthenticationService {
         authentication_server: &AuthenticationServerInfo,
         oidc_metadata: VerifiedClientMetadata,
     ) -> bool {
-        let issuer = &authentication_server.issuer;
+        let Ok(issuer) = Url::parse(&authentication_server.issuer) else {
+            tracing::error!("Failed to parse {:?}", authentication_server.issuer);
+            return false;
+        };
         let Some(registrations) = OidcRegistrations::new(
             &self.base_path,
-            self.oidc_configuration
-                .as_ref()
-                .map(|c| c.static_registrations.clone())
-                .unwrap_or_default(),
+            oidc_metadata.clone(),
+            self.oidc_static_registrations(),
         )
         .ok() else {
             return false;
         };
-        let Some(client_id) = registrations.client_id(issuer) else {
+        let Some(client_id) = registrations.client_id(&issuer) else {
             return false;
         };
 
         let client_data = RegisteredClientData {
-            credentials: ClientCredentials::None { client_id },
+            credentials: ClientCredentials::None { client_id: client_id.to_string() },
```

Comment:
Note: this `to_string()` can be replaced by taking ownership of the underlying string:

```suggestion
            credentials: ClientCredentials::None { client_id: client_id.0 },
```